### PR TITLE
parser: parse kill statement

### PR DESCRIPTION
**Pull request by tiancaiamao**
Parser support for kill statement https://dev.mysql.com/doc/refman/5.7/en/kill.html:

```
kill [query | connection] sessionID
```

This will be used for cancel session feature https://github.com/pingcap/tidb/issues/2528, together with the`show processlist` statement.

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -6545,4 +6546,44 @@ TableLockList:
...
+			TiDBExtension: $1.(bool),
+		}
+	}
+
+KillOrKillTiDB:
+	"KILL"
+	{
+		$$ = false
+	}
+|	"KILL" "TIDB"
```
Comment (tiancaiamao) [inline parser/parser.y]: `KILL TIDB XXX` is a grammar extension in TiDB.

Comment (coocood) [inline parser/parser.y]: Add this comment in source code.

File: ast/misc.go [hunk 2]
```diff
@@ -313,6 +314,27 @@ func (n *FlushStmt) Accept(v Visitor) (Node, bool) {
...
 
+// KillStmt is a statement to kill a query or connection.
+type KillStmt struct {
+	stmtNode
+
+	// If Query is true, terminates the statement the connection is currently executing, but leaves the connection itself intact.
+	// If Query is false, terminates the connection associated with the given ConnectionID, after terminating any statement the connection is executing.
+	Query         bool
+	ConnectionID  uint64
+	TiDBExtension bool
```
Comment (hanfei1991) [inline ast/misc.go]: What is this field used for ?

**Comment by tiancaiamao** [comment]
PTAL @hanfei1991 @coocood

**Comment by hanfei1991** [comment]
TiDB

**Comment by hanfei1991** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**